

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en-US" > <![endif]-->
<!--[if gt IE 8]><!-->
<html class="no-js" lang="en-US"> <!--<![endif]-->

<head>
  <!-- cookie consent -->
  
    <!-- Combined Onetrust and Rudderstack Implementation Scripts -->
    <!-- Onetrust Initialization -->
    <script type="text/javascript" src="https://cdn.cookielaw.org/consent/92466579-1717-44d3-809d-a05fb02843ed-test/OtAutoBlock.js"></script>
    <script src="https://cdn.cookielaw.org/scripttemplates/otSDKStub.js" data-document-language="true" type="text/javascript" charset="UTF-8" data-domain-script="92466579-1717-44d3-809d-a05fb02843ed-test"></script>
    <link rel="stylesheet" id="db-onetrust-style" href="https://www.databricks.com/wp-content/uploads/db_onetrust.css" media="all" />
    <!-- Setting Rudderstack Write Key -->
    <script>window.rudderstackKey = "2SOR9fvSr5Fi6tN2ihPbVHnX1SZ" </script>
    <!-- Rudderstack Initialization + Onetrust Integration + Rudderstack Custom Events -->
    <script type="text/javascript" src="https://www.databricks.com/sites/default/files/rudderstack/v1/db-rudderstack-events.js"></script>

  <!-- cookie consent -->

  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta http-equiv="X-UA-Compatible" content="IE=9" />
  <meta content="Learn about using the joint genotyping pipeline contained in Databricks Runtime for Genomics." name="description" />
<meta content="noindex" name="robots" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no" />
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=0">
  <meta property="og:image" content="https://www.databricks.com/wp-content/uploads/2020/04/og-databricks.png">
  <meta property="og:image:type" content="image/png">
  <meta property="og:title" content="Joint genotyping pipeline">
  <meta property="og:image:width" content="1200">
  <meta property="og:image:height" content="630">
  <meta property="og:type" content="website">
  <meta property="og:url" content="https://docs.databricks.com">
  <meta property="og:description" content="" id="og-description">
  <meta name="twitter:image" content="https://www.databricks.com/wp-content/uploads/2020/04/og-databricks.png">
  <meta name="twitter:site" content="@databricks">
  <meta name="twitter:creator" content="@databricks">
  <meta property="twitter:description" content="">
  
  <title>Joint genotyping pipeline &#124; Databricks on AWS</title>
  
  
  <link rel="canonical" href="https://docs.databricks.com/en/archive/genomics/joint-genotyping/joint-genotyping-pipeline.html">
  <!-- Start hreflang tag -->
  <link rel="alternate" hreflang="en" href="https://docs.databricks.com/en/archive/genomics/joint-genotyping/joint-genotyping-pipeline.html" />
<link rel="alternate" hreflang="x-default" href="https://docs.databricks.com/en/archive/genomics/joint-genotyping/joint-genotyping-pipeline.html" />
  <!-- End hreflang tag -->
  
  
  <link rel="shortcut icon" href="../../../_static/favicon.ico" />
  

  

  

  <!-- Google Tag Manager -->
  <script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
'https://www.googletagmanager.com/gtm.js?id='+i+dl;
j.setAttributeNode(d.createAttribute('data-ot-ignore'));
f.parentNode.insertBefore(j,f);
})(window,document,'script','dataLayer','GTM-T85FQ33');</script>
  <script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
'https://www.googletagmanager.com/gtm.js?id='+i+dl;
j.setAttributeNode(d.createAttribute('data-ot-ignore'));
f.parentNode.insertBefore(j,f);
})(window,document,'script','dataLayer','GTM-TWTKQQ');</script>
    
  <!-- End Google Tag Manager -->


  <!-- MaxMind / GEO IP -->
  <script src="//js.maxmind.com/js/apis/geoip2/v2.1/geoip2.js" type="text/javascript"></script>
  <!-- End MaxMind / GEO IP -->

  
  
  <link href="https://fonts.googleapis.com/css?family=Source+Sans+Pro:300,400,600&display=swap" rel="stylesheet">
  <link rel="preload" href="../../../_static/fonts/DMSans-Bold.ttf" as="font">
  <link rel="preload" href="../../../_static/fonts/DMSans-Regular.ttf" as="font">
  <link rel="preload" href="../../../_static/fonts/DMMono-Regular.ttf" as="font">
  <link rel="stylesheet" href="../../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../../_static/css/custom.css" type="text/css" />
  <link rel="stylesheet" href="../../../_static/css/cloud-provider-selector.css" type="text/css" />
  <link rel="stylesheet" href="../../../_static/css/translation-selector.css" type="text/css" />
  <link rel="stylesheet" href="../../../_static/css/searchunify/main.css" type="text/css" />

  
  <link rel="index" title="Index" href="../../../genindex.html" />
  <link rel="search" title="Search" href="../../../search.html" />
  <link rel="top" title="Databricks on AWS" href="../../../index.html" /> 
</head>

<body class="wy-body-for-nav" role="document">

  <!-- Google Tag Manager (noscript) -->
  <noscript><iframe src="https://www.googletagmanager.com/ns.html?id=GTM-T85FQ33"
height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
  <noscript><iframe src="https://www.googletagmanager.com/ns.html?id=GTM-TWTKQQ"
    height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
  <!-- End Google Tag Manager (noscript) -->

  
  <nav class="wy-nav-top header su_header" role="navigation" aria-label="top navigation">
    
<nav class="wy-nav-top header su_header" role="navigation" aria-label="top navigation">
  <div class="container-logo">
    <ul class="mobile-menu-toggle">
        <li class="menu-toggle">
            <i data-toggle="wy-nav-top" class="wy-nav-top-menu-button db-icon db-icon-menu pull-left"></i>
            
            <a href="https://www.databricks.com/" class="wy-nav-top-logo"><img src="../../../_static/small-scale-lockup-full-color-rgb.svg" width="137" height="21"
              alt="Databricks" /></a>   
               
              </li>
    </ul>
    <ul class="su_nav-menu">
      <li class="menu-toggle">
        <i data-toggle="wy-nav-top" class="wy-nav-top-menu-button db-icon db-icon-menu pull-left"></i>
        
          
        
        <a href="https://www.databricks.com/" class="wy-nav-top-logo"><img src="../../../_static/small-scale-lockup-full-color-rgb.svg" width="137" height="21"
            alt="Databricks" /></a></li>
        <!-- 
<li><a href="https://help.databricks.com/s/">Help Center</a></li>
<li class="active"><a href="https://docs.databricks.com/en/">Documentation</a></li>
<li><a href="https://kb.databricks.com/">Knowledge Base</a></li>
 -->
    </ul>
  </div>
  <div class="su_nav-right">
    <ul class="su_link-mobile">
  <!-- Mobile header code can go here -->
</ul>
<ul class="right-try-list">
   
</ul>
  </div>
</nav>
  </nav>

  <div class="su_sub-header">
    <div class="container">
      <div class="su_sub-header-inner">
        <!-- <div class="su_subnav-menu-right">
  <div id="auto" style="width: 100%;">
    <div ng-controller="SearchautoController">
      <div bind-html-compile="autocompleteHtml">
        <form class="su__search-box-1" disabled="disabled">
          <input class="su__search-input" type="search" name="Search box" id="su__search-b" placeholder="Search Documentation" disabled="disabled"/>
          <button class="su__search-button" type="submit" class="button button-success" disabled="disabled">
            <svg width="24" height="24" viewBox="0 0 24 24">
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                fill="#333"></path>
            </svg>
          </button>
        </form>
      </div>
    </div>
  </div>
</div> -->
        <div class="search-lng-gap"></div>
        <div style="margin-left: 16px; margin-right: 16px;">
          <!-- <select name="lng selector" id="lng-selector">
    <option value="../../../../en/archive/genomics/joint-genotyping/joint-genotyping-pipeline.html" class="notranslate">English</option>
    <option value="../../../../ja/archive/genomics/joint-genotyping/joint-genotyping-pipeline.html" class="notranslate">日本語</option>
    <option value="../../../../pt/archive/genomics/joint-genotyping/joint-genotyping-pipeline.html" class="notranslate">Português (Brasil)</option>
</select> -->
        </div>
        <div class="cloud-selector-container">
          <!-- <select name="cloud provider selector" id="cloud-provider-selector">
    <option value="aws" selected class="notranslate">
        Amazon Web Services
    </option>
    <option value="azure"  class="notranslate">
        Microsoft Azure
    </option>
    <option value="gcp"  class="notranslate">
        Google Cloud Platform
    </option>
</select> -->
        </div>
      </div>
    </div>
  </div>
  <page class="js-page-container">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side su_nav-side">
<div class="wy-side-scroll">
  <div class="wy-side-nav-search">
    

    

    

    
  </div>

  <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
    
      <a href="../../../index.html" class="main-navigation-home">Databricks on AWS</a>
    

    
      

      
        <p class="caption"><span class="caption-text">Load &amp; manage data</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../rag-temp/index.html">RAG Studio</a></li>
</ul>

      
    
  </div>

  <div role="contentinfo">
    
  <p class="build_info notranslate"data-last-edit="December 23, 2023">
    Updated Jan 11, 2024
  </p>
<script>
  window.addEventListener('DOMContentLoaded',function(){
    var h1=document.querySelector('h1');
    var bi=document.querySelector('[data-last-edit]');
    if(h1 && bi){
      var ver = document.createElement('p');
      ver.className = 'version_info';
      ver.textContent = bi.getAttribute('data-last-edit');
      h1.parentElement.insertBefore(ver, h1.nextElementSibling);
    }
  });
</script>

    <p>
      
        <a id='feedbacklink' href="mailto:doc-feedback@databricks.com?subject=Documentation Feedback">Send us feedback</a>
      
    </p>
  </div>
</div>
</nav>
    
    
<main class="wy-grid-for-nav su_nav-grid">
  <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
    <div class="wy-nav-content su__nav_content">
      <div class="rst-content">
        





<div role="navigation" aria-label="breadcrumbs navigation" class="wy-breadcrumbs-wrapper">
  <ul class="wy-breadcrumbs">
    <li><a href="../../../index.html">Documentation</a> <span class="db-icon db-icon-chevron-right"></span></li>
    
    
      <li>Joint genotyping pipeline</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
      </li>
    
  </ul>
</div>
        
        <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
          <div itemprop="articleBody">
            
    
  <div class="section" id="joint-genotyping-pipeline">
<h1>Joint genotyping pipeline<a class="headerlink" href="#joint-genotyping-pipeline" title="Permalink to this headline"> </a></h1>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>This documentation has been retired and might not be updated. The products, services, or technologies mentioned in this content are no longer supported.</p>
<p>The Databricks Genomics runtime has been deprecated. For open source equivalents, see repos for <a class="reference external" href="https://github.com/databricks/genomics-pipelines">genomics-pipelines</a> and <a class="reference external" href="https://github.com/projectglow/glow">Glow</a>. Bioinformatics libraries that were part of the runtime have been released as a Docker container, which can be pulled from the <a class="reference external" href="https://hub.docker.com/r/projectglow">ProjectGlow Dockerhub</a> page.</p>
<p>For more information about the Databricks Runtime deprecation policy and schedule, see <a class="reference internal" href="../../../release-notes/runtime/index.html#supported-list"><span class="std std-ref">All supported Databricks Runtime releases</span></a>.</p>
</div>
<p>The Databricks joint genotyping pipeline is a <a class="reference external" href="https://gatk.broadinstitute.org/hc/en-us/articles/360035535932-Germline-short-variant-discovery-SNPs-Indels-">GATK best practices</a> compliant pipeline for joint genotyping
using <a class="reference external" href="https://gatk.broadinstitute.org/hc/en-us/articles/360041417092-GenotypeGVCFs">GenotypeGVCFs</a>.</p>
<div class="section" id="walkthrough">
<h2>Walkthrough<a class="headerlink" href="#walkthrough" title="Permalink to this headline"> </a></h2>
<p>The pipeline typically consists of the following steps:</p>
<ol class="arabic simple">
<li><p>Ingest variants into Delta Lake.</p></li>
<li><p>Joint-call the cohort with GenotypeGVCFs.</p></li>
</ol>
<p>During variant ingest, single-sample gVCFs are processed in batches and the rows are stored in Delta Lake to provide fault
tolerance, fast querying, and incremental joint genotyping. In the joint genotyping step, the gVCF rows are ingested
from Delta Lake, split into bins, and distributed to partitions. For each variant site, the relevant gVCF rows per sample are
identified and used for regenotyping.</p>
</div>
<div class="section" id="setup">
<h2>Setup<a class="headerlink" href="#setup" title="Permalink to this headline"> </a></h2>
<p>The pipeline is run as a Databricks job. Most likely a Databricks solutions architect will work with
you to set up the initial job. The necessary details are:</p>
<div class="highlight-json notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
<span class="w">  </span><span class="nt">&quot;autoscale.min_workers&quot;</span><span class="p">:</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nt">&quot;type&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;unlimited&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="nt">&quot;defaultValue&quot;</span><span class="p">:</span><span class="w"> </span><span class="mi">1</span>
<span class="w">  </span><span class="p">},</span>
<span class="w">  </span><span class="nt">&quot;autoscale.max_workers&quot;</span><span class="p">:</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nt">&quot;type&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;unlimited&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="nt">&quot;defaultValue&quot;</span><span class="p">:</span><span class="w"> </span><span class="mi">25</span>
<span class="w">  </span><span class="p">},</span>
<span class="w">  </span><span class="nt">&quot;enable_elastic_disk&quot;</span><span class="p">:</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nt">&quot;type&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;fixed&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="nt">&quot;value&quot;</span><span class="p">:</span><span class="w"> </span><span class="kc">true</span>
<span class="w">  </span><span class="p">},</span>
<span class="w">  </span><span class="nt">&quot;node_type_id&quot;</span><span class="p">:</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nt">&quot;type&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;unlimited&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="nt">&quot;defaultValue&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;i3.8xlarge&quot;</span>
<span class="w">  </span><span class="p">},</span>
<span class="w">  </span><span class="nt">&quot;spark_env_vars.refGenomeId&quot;</span><span class="p">:</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nt">&quot;type&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;unlimited&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="nt">&quot;defaultValue&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;grch38&quot;</span>
<span class="w">  </span><span class="p">},</span>
<span class="w">  </span><span class="nt">&quot;spark_version&quot;</span><span class="p">:</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nt">&quot;type&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;regex&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="nt">&quot;pattern&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;.*-hls.*&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="nt">&quot;defaultValue&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;7.4.x-hls-scala2.12&quot;</span>
<span class="w">  </span><span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
<ul class="simple">
<li><p>The cluster configuration should use Databricks Runtime for Genomics.</p></li>
<li><p>The task should be the joint genotyping pipeline notebook found at the bottom of this page.</p></li>
<li><p>For best performance, use storage-optimized instances. We recommend <strong>i3.8xlarge</strong>.</p></li>
<li><p>To reduce costs, use all spot workers with the <strong>Spot fall back to On-demand</strong> option selected.</p></li>
<li><p>To reduce costs, enable autoscaling with a minimum of 1 worker and a maximum of 10-50 depending
on latency requirements.</p></li>
<li><p>Enable autoscaling local storage to ensure that the cluster doesn’t run out of disk space</p></li>
</ul>
</div>
<div class="section" id="reference-genomes">
<h2>Reference genomes<a class="headerlink" href="#reference-genomes" title="Permalink to this headline"> </a></h2>
<p>You must configure the reference genome using <a class="reference internal" href="../../../compute/configure.html#env-var"><span class="std std-ref">environment variables</span></a>.
To use GRCh37, set the environment variable:</p>
<div class="highlight-ini notranslate"><div class="highlight"><pre><span></span><span class="na">refGenomeId</span><span class="o">=</span><span class="s">grch37</span>
</pre></div>
</div>
<p>To use GRCh38, change <code class="docutils literal notranslate"><span class="pre">grch37</span></code> to <code class="docutils literal notranslate"><span class="pre">grch38</span></code>.</p>
<p>To use a custom reference genome, see instructions in <a class="reference internal" href="../secondary/dnaseq-pipeline.html#custom-reference-genomes"><span class="std std-ref">Custom reference genomes</span></a>.</p>
</div>
<div class="section" id="parameters">
<h2>Parameters<a class="headerlink" href="#parameters" title="Permalink to this headline"> </a></h2>
<p>The pipeline accepts parameters that control its behavior. The most important and
commonly changed parameters are documented here. To view all available parameters and their usage
information, run the first cell of the pipeline notebook. New parameters are added regularly.
After importing the notebook and setting it as a job task, you can set these parameters for
<a class="reference internal" href="../../../workflows/jobs/create-run-jobs.html#create-a-job"><span class="std std-ref">all runs</span></a> or <a class="reference internal" href="../../../workflows/jobs/create-run-jobs.html#job-run-with-different-params"><span class="std std-ref">per-run</span></a>.</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Parameter</p></th>
<th class="head"><p>Default</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>manifest</p></td>
<td><p>n/a</p></td>
<td><p>The <a class="reference internal" href="#manifest-format"><span class="std std-ref">manifest</span></a> describing the input.</p></td>
</tr>
<tr class="row-odd"><td><p>output</p></td>
<td><p>n/a</p></td>
<td><p>The path where pipeline <a class="reference internal" href="#output"><span class="std std-ref">output</span></a> is written.</p></td>
</tr>
<tr class="row-even"><td><p>replayMode</p></td>
<td><p>skip</p></td>
<td><p>One of:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">skip</span></code>: stages are skipped if output already exists.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">overwrite</span></code>: existing output is deleted.</p></li>
</ul>
</td>
</tr>
<tr class="row-odd"><td><p>exportVCF</p></td>
<td><p>false</p></td>
<td><p>If true, the pipeline writes results in VCF as well as Delta Lake.</p></td>
</tr>
<tr class="row-even"><td><p>targetedRegions</p></td>
<td><p>n/a</p></td>
<td><p>Path to files containing regions to call. If omitted, calls all regions.</p></td>
</tr>
<tr class="row-odd"><td><p>gvcfDeltaOutput</p></td>
<td><p>n/a</p></td>
<td><p>If specified, gVCFs are ingested to a Delta table before genotyping. You should
specify this parameter only if you expect to joint call the same gVCFs many times.</p></td>
</tr>
<tr class="row-even"><td><p>performValidation</p></td>
<td><p>false</p></td>
<td><p>If <code class="docutils literal notranslate"><span class="pre">true</span></code>, the system verifies that each record contains the necessary information for
joint genotyping. In particular, it checks that the correct number of genotype
probabilities are present.</p></td>
</tr>
<tr class="row-odd"><td><p>validationStringency</p></td>
<td><p>STRICT</p></td>
<td><p>How to handle malformed records, both during loading and validation.</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">STRICT</span></code>: fail the job</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">LENIENT</span></code>: log a warning and drop the record</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">SILENT</span></code>: drop the record without a warning</p></li>
</ul>
</td>
</tr>
</tbody>
</table>
<div class="admonition tip" id="manifest-format">
<p class="admonition-title">Tip</p>
<p>To perform joint calling from an existing Delta table, set <code class="docutils literal notranslate"><span class="pre">gvcfDeltaOutput</span></code> to the table path and <code class="docutils literal notranslate"><span class="pre">replayMode</span></code> to <code class="docutils literal notranslate"><span class="pre">skip</span></code>. You can also provide the <code class="docutils literal notranslate"><span class="pre">manifest</span></code>, which will be used to define the VCF schema and samples; these will be inferred from the Delta table otherwise. We ignore the <code class="docutils literal notranslate"><span class="pre">targetedRegions</span></code> and <code class="docutils literal notranslate"><span class="pre">performValidation</span></code> parameters in this setup.</p>
</div>
</div>
<div class="section" id="output">
<h2>Output<a class="headerlink" href="#output" title="Permalink to this headline"> </a></h2>
<p>The regenotyped variants are all written out to Delta tables inside the provided output directory. In addition, if
you configured the pipeline to export VCFs, they’ll appear under the output directory as well.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">output</span>
<span class="o">|---</span><span class="n">genotypes</span>
    <span class="o">|---</span><span class="n">Delta</span> <span class="n">files</span>
<span class="o">|---</span><span class="n">genotypes</span><span class="o">.</span><span class="n">vcf</span>
    <span class="o">|---</span><span class="n">VCF</span> <span class="n">files</span>
</pre></div>
</div>
</div>
<div class="section" id="manifest-format">
<span id="manifest-format-1"></span><h2>Manifest format<a class="headerlink" href="#manifest-format" title="Permalink to this headline"> </a></h2>
<p>The manifest is a file or blob describing where to find the input single-sample GVCF files, with each file path on a new
row. For example:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>HG00096.g.vcf.bgz
HG00097.g.vcf.bgz
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>If the provided manifest is a file, each row may be an absolute path or a path relative to the manifest file. If the
provided manifest is a blob, the row field must be an absolute path. You can include globs <code class="docutils literal notranslate"><span class="pre">(*)</span></code> to match many files.</p>
</div>
</div>
<div class="section" id="troubleshooting">
<h2>Troubleshooting<a class="headerlink" href="#troubleshooting" title="Permalink to this headline"> </a></h2>
<p><strong>Job fails with an <code class="docutils literal notranslate"><span class="pre">ArrayIndexOutOfBoundsException</span></code></strong></p>
<p>This error usually indicates that an input record has an incorrect number of genotype probabilities. Try setting the <code class="docutils literal notranslate"><span class="pre">performValidation</span></code> option to <code class="docutils literal notranslate"><span class="pre">true</span></code> and the <code class="docutils literal notranslate"><span class="pre">validationStringency</span></code> option to <code class="docutils literal notranslate"><span class="pre">LENIENT</span></code> or <code class="docutils literal notranslate"><span class="pre">SILENT</span></code>.</p>
</div>
<div class="section" id="additional-usage-info">
<h2>Additional usage info<a class="headerlink" href="#additional-usage-info" title="Permalink to this headline"> </a></h2>
<p>The joint genotyping pipeline shares many operational details with the other Databricks pipelines. For
more detailed usage information, such as output format structure, tips for running programmatically,
and steps for setting up custom reference genomes, see <a class="reference internal" href="../secondary/dnaseq-pipeline.html"><span class="doc">DNASeq pipeline</span></a>.</p>
</div>
</div>


    
          </div>
        </div>
        <div  class="suapp-rating">
  <div id="suPageRateApp">
     <su-app></su-app>
   </div> 
 </div>
<hr> 
<footer>
  <div role="contentinfo">
      <p class="copyright">
          &copy; Databricks 2023. All rights reserved. Apache, Apache Spark, Spark, and the Spark logo are trademarks of the <a href="http://www.apache.org/">Apache Software Foundation</a>.
      </p>
      <p> 
        
          <a id='feedbacklink' href="mailto:doc-feedback@databricks.com?subject=Documentation Feedback">Send us feedback</a>
        
     | <a href="https://databricks.com/privacy-policy">Privacy Policy</a> | <a href="https://databricks.com/terms-of-use">Terms of Use</a></p>

  </div> 

</footer>
      </div>
    </div>
  </section>
</main>

  </page>
  
  <script type="text/javascript">
    var DOCUMENTATION_OPTIONS = {
      URL_ROOT: '../../../',
      VERSION: '1.0',
      COLLAPSE_INDEX: false,
      FILE_SUFFIX: '.html',
      HAS_SOURCE: 'false'
    };
  </script>
  <script type="text/javascript" src="../../../_static/jquery.js"></script>
  <script type="text/javascript" src="../../../_static/underscore.js"></script>
  <script type="text/javascript" src="../../../_static/doctools.js"></script>
  <script type="text/javascript" src="../../../_static/language_data.js"></script>
  

  <script type="text/javascript" src="../../../_static/js/clipboard.min.js"></script>
  <script type="text/javascript" src="../../../_static/js/jquery.waypoints.min.js"></script>

  <!-- Select2 (https://select2.org/) -->
  <link href="https://cdn.jsdelivr.net/npm/select2@4.1.0-rc.0/dist/css/select2.min.css" rel="stylesheet" />
  <script src="https://cdn.jsdelivr.net/npm/select2@4.1.0-rc.0/dist/js/select2.min.js"></script>
  <!-- End Select2 -->

  
  
  <script type="text/javascript" src="../../../_static/js/localized.js"></script>
  <script type="text/javascript" src="../../../_static/js/custom.js"></script>
  

  
  
  <script type="text/javascript">
    jQuery(function () {
      SphinxRtdTheme.StickyNav.enable();
    });

  </script>
  
 



  <script>
  window.__searchunifyLoaderConfig = JSON.parse('{"clients": {"en": "02c2e804-27e9-11ee-aefb-0242ac120011", "ja": "6a42c3f2-2820-11ee-aefb-0242ac120011", "pt": "6a86badd-2821-11ee-aefb-0242ac120011"}}')
</script>
<script type="text/javascript" src="../../../_static/js/search-loader.js"></script>
</body>
<script type='text/javascript'>
  window.onload = function () {
    var description = document.querySelector('meta[name="description"]').getAttribute("content");
    let titleText = document.querySelector('h1').textContent;
    document.querySelector('meta[property="og:title"]').setAttribute("content", titleText);
    document.querySelector('meta[property="og:description"]').setAttribute("content", description);
    document.querySelector('meta[property="twitter:description"]').setAttribute("content", description);
  };
</script>

</html>